pkg/controllers/chaperon/controller.go

```diff
@@ -214,3 +225,15 @@ func newPod(podChaperon *multiclusterv1alpha1.PodChaperon) *corev1.Pod {
 	podChaperon.Spec.DeepCopyInto(&pod.Spec)
 	return pod
 }
+
+func podNeverExisted(podChaperon *multiclusterv1alpha1.PodChaperon) bool {
+	if podChaperon.Status.Phase == "" {
+		return true
+	}
+	for _, condition := range podChaperon.Status.Conditions {
```

Comment:
I don't understand. If the candidate pod creation failed, why would podChaperon.Status.Phase not be empty?

Reply:
So prior to this change, the following will happen:

1. Attempt to create the pod.
https://github.com/admiraltyio/admiralty/blob/f98c59d93af1f3908f4a27dd4071fd5c486a863d/pkg/controllers/chaperon/controller.go#L120
2. Pod creation fails.
https://github.com/admiraltyio/admiralty/blob/f98c59d93af1f3908f4a27dd4071fd5c486a863d/pkg/controllers/chaperon/controller.go#L122 and method returns
3. The status setting code on the chaperon never gets to run because of the early exit above so the chaperon status is never updated.
https://github.com/admiraltyio/admiralty/blob/f98c59d93af1f3908f4a27dd4071fd5c486a863d/pkg/controllers/chaperon/controller.go#L158

Reply:
After this change, only the condition is updated, so the phase should still be empty, right? So line 235 (return true) is never executed. Checking the condition here appears to be unnecessary.

Reply:
woops I misunderstood your earlier question, good catch! I think it would be reasonable to set the phase to pending in that failed case since `Failed` implies container termination.